src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -223,32 +223,48 @@ ImmutableArray<ExpressionSyntax> assignments
 					)
 				);
 
-				immutable = false;
+				// Note: we're going to go looking for other errors.
+				// There shouldn't be any "return true" below, it should
+				// always be conditional on isReadOnly.
 			}
 
-			// Perform an initial check to see if the member is totally immutable
-			// If this fails, then we need to perform assignment checks to ensure
-			// instance immutability
+			// If our field or property is a type that is always immutable then
+			// we can stop looking (all that matters is that we are readonly).
 			if( m_context.IsImmutable(
 				type,
 				ImmutableTypeKind.Total,
 				typeSyntax.GetLocation,
-				out _
+				out var typeDiagnostic
 			) ) {
-				return immutable;
+				return isReadOnly;
 			}
 
-			// Check all assignments of the member for immutability
+			// Our field or property could hold mutable values, but it's safe
+			// as long as we always assign immutable values to it.
+
+			// Usually that would be difficult, but if we're readonly its a lot
+			// easier: we only have to look in our constructors (this applies
+			// even to protected readonly; these can't be set in derived
+			// constructors.)
+
+			// TODO: we don't really need this, but it has been our current
+			// behaviour... if we have a readonly field with no writes that's
+			// worth a diagnostic on its own (there are built in ones for that)
+			// but technically it doesn't add mutability.
+			// If we remove this we can rename typeDiagnostic to _ again.

```

Comment:
https://github.com/Brightspace/D2L.CodeStyle/issues/721

Reply:
The `Any()` is trash but I can fix this before deploying.

Reply:
Any hard reason to not just remove this and close #721?

Reply:
yeah, the second bit -- just wanted to keep this PR smaller.

Reply:
Like it will require changing the tests 😬